lib/DDG/Spice/TodayInHistory.pm

```diff
@@ -3,33 +3,83 @@ package DDG::Spice::TodayInHistory;
 
 use strict;
 use DDG::Spice;
+use DateTime;
+use DateTime::TimeZone;
+use Time::Piece;
 
-name 'Today In History';
-description 'this day in history';
-source 'History.com';
-primary_example_queries 'today in history';
-secondary_example_queries 'this day in history';
-category 'facts';
-topics 'everyday', 'special_interest';
-icon_url 'http://www.history.com/favicon.ico';
-code_url 'https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/lib/DDG/Spice/TodayInHistory.pm';
-attribution web => [ 'https://www.duckduckgo.com', 'DuckDuckGo' ],
-            github => [ 'https://github.com/duckduckgo', 'duckduckgo'],
-            twitter => ['http://twitter.com/duckduckgo', 'duckduckgo'];
-
-triggers startend => (
-    'today in history',
-    'todays in history',
-    'this day in history',
-);
-
-spice to => 'http://www.history.com/this-day-in-history/rss';
-
-spice wrap_string_callback => 1;
+name "TodayInHistory";
+source "http://wikipedia.com";
+description "List of events which occured on this day in history";
+primary_example_queries "today in history", "this day in history";
+secondary_example_queries "historical events on 27 June";
+category "dates";
+topics "everyday","trivia";
+code_url "https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/lib/DDG/Spice/TodayInHistory.pm";
+attribution github => ["https://github.com/killerfish", "Usman Raza"],
+            twitter => ["https://twitter.com/f1shie", "Usman Raza"];
+
+spice to => 'http://en.wikipedia.org/w/api.php?action=query&titles=$1&prop=revisions&rvlimit=1&rvprop=content&format=json&continue=&redirects&callback={{callback}}';
+
+triggers startend => "today in history";
+triggers start => "historical events on", "this day in history";
+
+# define variables
+my %days = ('1'=>'31','2'=>'29','3'=>'31','4'=>'30','5'=>'31','6'=>'30','7'=>'31','8'=>'31','9'=>'30','10'=>'31','11'=>'30','12'=>'31');
+
+# functions:
+# check and process if we got 2 parameters (month and day)
+sub readParams {
+  # read parameters - do not trigger if invalid parameters are found (delimiters: - / . ' ')
+    my ($parameter1, $parameter2) = split (/[-\/. ]/, $_[0]);
+    if ($parameter2) {
+    my $value;
+        # e.g. "historical events on June 12"
+        return $value if (($value = validateParams($parameter1, $parameter2)) ne 'fail');
+        # e.g. "historical events on 16 Sep"
+        return $value if (($value = validateParams($parameter2, $parameter1)) ne 'fail');
+    }
+    # no parameter - use today date
+    else {
+        return useToday();
+    }
+}
+
+# validate parameters and return day_month on success
+sub validateParams {
```

Comment:
Can we use https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/lib/DDG/GoodieRole/Dates.pm here instead? But only change if it will solve issues like https://github.com/duckduckgo/zeroclickinfo-spice/pull/1988/files?diff=split#r33841410 for us and none of this is problematic (like we'd probably want support for "jul 10" as well). If it's not any better it's best to come back to it